packages/server/lib/controllers/proxy.controller.ts

```diff
@@ -425,38 +440,39 @@ class ProxyController {
                 { numOfAttempts: Number(config.retries), retry: proxyService.retry.bind(this, config, logs) }
             );
 
-            // We batch save, since we have buffered the createdAt it shouldn't impact order
-            await Promise.all(
-                logs.map(async (log) => {
-                    if (log.level === 'debug' && !isDebug) {
-                        return;
-                    }
-                    await logCtx.log(log);
-                })
-            );
+            await flushLogsBuffer(logs, logCtx);
 
-            await this.handleResponse({ res, responseStream, config, url, logCtx });
+            await this.handleResponse({ res, responseStream, config, requestConfig, logCtx });
         } catch (error) {
-            await this.handleErrorResponse(res, error, url, config, logCtx);
+            await this.handleErrorResponse({ res, e: error, requestConfig, config, logCtx });
             metrics.increment(metrics.Types.PROXY_FAILURE);
         }
     }
 
-    private async reportError(
-        error: AxiosError,
-        url: string,
-        config: ApplicationConstructedProxyConfiguration,
-        errorContent: string | Record<string, string>,
-        logCtx: LogContext
-    ) {
-        const safeHeaders = proxyService.stripSensitiveHeaders(config.headers, config);
-        await logCtx.http(`${config.method.toUpperCase()} ${url} failed with status '${error.response?.status}'`, {
+    private async reportError({
+        error,
+        config,
+        requestConfig,
+        errorContent,
+        logCtx
+    }: {
+        error: AxiosError;
+        config: ApplicationConstructedProxyConfiguration;
+        requestConfig: AxiosRequestConfig;
+        errorContent: string | Record<string, string>;
+        logCtx: LogContext;
+    }) {
+        const valuesToFilter = Object.values(config.connection.credentials);
+        const safeHeaders = redactHeaders({ headers: requestConfig.headers, valuesToFilter });
+        const redactedURL = redactURL({ url: requestConfig.url!, valuesToFilter });
```

Comment:
While doing that I realized it's still suboptimal. A more bulletproof approach would be to set a redaction strategy in the LogContext that would redact the JSON.stringify string just before sending it to elasticsearch. This could lead to false positives but less sweat and mistakes

Reply:
that would be nicer indeed and would require to think about redaction when using `logCtx.http()`